src/triage/experiments/base.py

```diff
@@ -322,10 +333,17 @@ def collate_aggregations(self):
 
         """
         logging.info('Creating collate aggregations')
+        if hasattr(self, 'state_table_generator'):
+            cohort_table = self.state_table_generator.sparse_table_name
+        else:
+            cohort_table = None
```

Comment:
seems like it would be easier and less surprising if the class always defined its attribute `state_table_generator`, but that this were in the fallback/warning case set to `None`.

and then it would be especially easy and concise to just retrieve the attribute you want here (and elsewhere), rather than try to access it more than once:

    cohort_table = getattr(self.state_table_generator, 'sparse_table_name', None)

Reply:
(or for that matter it could fallback to a dummy state table generator, with stubbed-out methods like `clean_up`, and a `sparse_table_name` set to `None` -- that way you needn't spread around the code logic to handle this case.)

Reply:
The first version of this did have a null object `NoOpStateTableGenerator`, but in practice I didn't like it. Basically it would allow the experiment to go too far without short-circuiting, which hurts the experience for the user because there's a bunch of no-op logging after the experiment has finished its useful work.

I'm not saying that it would be impossible to make work with a null object pattern, but I think there has to be more thought put into preventing no-op log spam using that method which I haven't been able to figure out, and that's how I arrived at this.